src/test/java/com/microsoft/sqlserver/jdbc/AlwaysEncrypted/AESetup.java

```diff
@@ -0,0 +1,158 @@
+import java.util.Properties;
+
+import javax.xml.bind.DatatypeConverter;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerColumnEncryptionJavaKeyStoreProvider;
+import com.microsoft.sqlserver.jdbc.SQLServerColumnEncryptionKeyStoreProvider;
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Utils;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@RunWith(JUnitPlatform.class)
+public class AESetup extends AbstractTest {
+
+    static String javaKeyStoreInputFile = "JavaKeyStore.txt";
+    static String keyStoreName = "MSSQL_JAVA_KEYSTORE";
+    static String jksName = "clientcert.jks";
+    static String filePath = null;
+    static String thumbprint = null;
+    static SQLServerConnection con = null;
+    static Statement stmt = null;
+    static String cmkName = "JDBC_CMK";
+    static String cekName = "JDBC_CEK";
+    static String keyPath = null;
+    static String certStore = null;
+    static String javaKeyAliases = null;
+    static String OS = System.getProperty("os.name").toLowerCase();
+    static SQLServerColumnEncryptionKeyStoreProvider storeProvider = null;
+    static String secretstrJks = "password";
+    static String numericTable = "numericTable";
+
+    /**
+     * Create connection, statement and generate path of resource file
+     * 
+     * @throws SQLException
+     */
+    @BeforeAll
+    static void setUpConnection() throws SQLException {
+        readFromFile(javaKeyStoreInputFile, "Alias name");
+        con = (SQLServerConnection) DriverManager.getConnection(connectionString);
+        stmt = con.createStatement();
+        Utils.dropTableIfExists(numericTable, stmt);
+        dropCEK();
+        dropCMK();
+        con.close();
+       
+
+        keyPath = Utils.getCurrentClassPath() + jksName;
+        storeProvider = new SQLServerColumnEncryptionJavaKeyStoreProvider(keyPath, secretstrJks.toCharArray());
+        Properties info = new Properties();
+        info.setProperty("ColumnEncryptionSetting", "Enabled");
+        info.setProperty("keyStoreAuthentication", "JavaKeyStorePassword");
+        info.setProperty("keyStoreLocation", keyPath);
+        info.setProperty("keyStoreSecret", secretstrJks);
+        con = (SQLServerConnection) DriverManager.getConnection(connectionString, info);
+        stmt = con.createStatement();
+        createCMK(keyStoreName, javaKeyAliases);
+        certStore = keyStoreName;        
+    }
+
+    private static void readFromFile(String inputFile,
+            String lookupValue) {
+        filePath = Utils.getCurrentClassPath();
+        try {
+            File f = new File(filePath + inputFile);
+            BufferedReader b = new BufferedReader(new FileReader(f));
```

Comment:
Move `BufferedReader b` before try block.  
In try block initialize it and close it in finally block.